src/test/java/seedu/address/logic/commands/AddProcCommandTest.java

```diff
@@ -1,214 +1,116 @@
+import static seedu.address.logic.commands.CommandTestUtil.VALID_DATE_REPAIR;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_HASCOMPLETED_TRUE;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_INFO_REPAIR;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
 import static seedu.address.testutil.Assert.assertThrows;
-
-import java.nio.file.Path;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.function.Predicate;
+import static seedu.address.testutil.TypicalClients.getTypicalAddressBook;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_CLIENT;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_CLIENT;
 
 import org.junit.jupiter.api.Test;
 
-import javafx.collections.ObservableList;
-import seedu.address.commons.core.GuiSettings;
-import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
 import seedu.address.model.AddressBook;
 import seedu.address.model.Model;
-import seedu.address.model.ReadOnlyAddressBook;
-import seedu.address.model.ReadOnlyUserPrefs;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
 import seedu.address.model.client.Client;
 import seedu.address.model.procedure.Procedure;
 import seedu.address.testutil.ClientBuilder;
+import seedu.address.testutil.ProcedureBuilder;
 
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for AddProcCommand.
+ */
 public class AddProcCommandTest {
 
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    // new Model without any Clients
+    private Model newModel = new ModelManager();
+    private Procedure toBeAddedProcedure = new ProcedureBuilder().build();
+
     @Test
-    public void constructor_nullClient_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> new AddCommand(null));
+    public void execute_properIndexWithProperProcedure_success() {
+        // guaranteed to add a non-duplicated Procedure since starting from a Client with no Procedures
+        Client newClient = new ClientBuilder().build();
+        newModel.addClient(newClient);
+
+        AddProcCommand addProcCommand = new AddProcCommand(INDEX_FIRST_CLIENT, toBeAddedProcedure);
+
+        String expectedMessage = String.format(AddProcCommand.MESSAGE_SUCCESS, toBeAddedProcedure);
+
+        Model expectedModel = new ModelManager(new AddressBook(newModel.getAddressBook()), new UserPrefs());
+        Client clientToAddProc = newModel.getFilteredClientList().get(0);
+        Client clientWithAddedProcedure = new ClientBuilder(clientToAddProc)
+                .withProcedures(addProcCommand.procListWithAddedProperProc(clientToAddProc.getProcedures())).build();
```

Comment:
Good abstraction thus far, but I think it would be good to separate these chunks of code to increase developer's readability :), what do you think?